### DISCUSSION TITLE
Add/Edit link in WYSIWYG, allow linking to files

### DISCUSSION THREAD

**Discussion by avatarbabe** [directus/directus#6605] [Feature Requests]
Similar to [https://github.com/directus/directus/issues/3160](https://github.com/directus/directus/issues/3160) it would be great if it was possible to link to a file in the directus file system from the Add/Edit link window in the RTE. Maybe I'm missing something, but currently you have to find the id of the file you want to link to and manually insert it in the Add/Edit link dialog along with the base url.

We need to link to several different PDF:s and currently it is very cumbersome to do so.

**Comment by mooxl** [comment]
We currently have a customer who would like to link PDFs uploaded to Directus in the WYSIWYG editor. Unfortunately, he is not as tech-savvy as we are, so it would be great if this feature could be implemented.

**Comment by mooxl** [comment]
I have just tried to implement this feature and I think we should first clarify where this linking should be possible.
There are three possibilities:

1. New element in the toolbar
2. New category under "Add/Edit Media"
3. "Choose File from Library"-Button in URL-Input at "Add/Edit Link"

In my opinion, the button at the URL-Input would make the most sense. 
What do you say?

**Comment by rijkvanzanten** [comment]
So to confirm, what you're looking for is a way to easily generate the full URL to a given file from the file library? So instead of inlining an image, you're essentially generating an `<a>` tag with the `href` pointing to the file?

**Reply by mooxl** [reply]
Exactly! Maybe there should also be an option to upload a file which will be linked to via an `<a>` tag. So like the upload/choose file process of the "Add/Edit Image" and the editing process of the "Add Link" module.

**Reply by rijkvanzanten** [reply]
> Maybe there should also be an option to upload a file which will be linked to via an <a> tag.

This is already (sort of) possible by selecting the file in the wysiwyg, and then using the existing link button to link it somewhere 🤔 

Now for your question, I'm not sure which of the three I'd prefer actually! Do you have any thoughts / preferences @mooxl?

**Reply by mooxl** [reply]
> This is already (sort of) possible by selecting the file in the wysiwyg, and then using the existing link button to link it somewhere

I meant that you link to the respective file and not include the file in an `<a>` tag and link to somewhere. Sorry, English is not my first language. 

If it were up to me I would implement my third suggestion, as the location of the feature seems the most logical to me and the option for uploading would just be a nice to have.

**Reply by rijkvanzanten** [reply]
> I meant that you link to the respective file and not include the file in an <a> tag and link  somewhere.

Yeah! I meant that you could already link to the file itself by adding a regular link, and then pasting the link to the file library. Definitely not as user friendly as you're suggesting, but _theoretically_ possible.

> Sorry, English is not my first language.

No worries! Same here 🇳🇱

> If it were up to me I would implement my third suggestion, as the location of the feature seems the most logical to me and the option for uploading would just be a nice to have.

Sounds good to me! Wanna take a swing at making a PR? 🙂

**Reply by mooxl** [reply]
> Yeah! I meant that you could already link to the file itself by adding a regular link, and then pasting the link to the file library. Definitely not as user friendly as you're suggesting, but _theoretically_ possible.

That's how we've done it with our customers so far. As you say, as soon as you have to explain it to someone who is not so tech-savvy, it becomes difficult:)

I'm pretty busy these days but will make time for it at the weekend!

**Reply by benhaynes** [reply]
Exciting, thank you!! ❤️

**Reply by mooxl** [reply]
Hey, unfortunately I didn't get around to finishing the first version of the PR this weekend. Working with Vue is more rusty than I thought at first:) That's why you'll have to be patient a little longer.

**Reply by joselcvarela** [reply]
No problem @mooxl. In the meantime if you already have something we can see it would be helpful just for guidance 👍

**Reply by AndreyKindin** [reply]
Is there anything new on this?

**Reply by mooxl** [reply]
no, sry

**Reply by FuEqu** [reply]
@mooxl hey, this is kind of an old thread, but is it possible to link to files from the file library by now? Thank you!

**Reply by narduin** [reply]
I have the same use case as everyone here.
I don't have the skills to dev this but I'll be glad to know if someone has!

**Comment by sepa-rater** [comment]
Yes, this particular function has gone down a step from v8, it was more intuitive for users to manage file links by adding or even select a file from library in the Add/Edit Link window.

The file library system overall is an improvement. However, I would like to see the ability easily manage file links right from the add/edit link window in wysiwyg editor.

I'll be keeping my eye on this, I may have to stay with v8 (for my users sake unfortunately) until such time.

**Comment by PeterSchoell** [comment]
Hello all,

we also have some customers who are very urgently waiting for this feature. In Directus 8 the file upload and link was available - with v9 this option has been removed. A separate menu item in the WYSIWYG editor is also a possibility, but from my point of view not necessary.
 
![image](https://user-images.githubusercontent.com/48986468/202528814-aa583142-e4e2-4c9f-a798-9ddee710d9fb.png)

The old function in the link dialog was sufficient and should definitely be integrated again.

**Comment by nervusdm** [comment]
Looking for the same feature here.

**Comment by Leapfrognz** [comment]
Also after this feature for a client

**Comment by nicksergeant** [comment]
Somewhat heartbroken after discovering there's no way for end users to easily add or choose an uploaded file in the WYSIWYG editor.

We build admin interfaces for non-technical folks and I landed on Directus because of how nice the API and UI is for end users.

Am I correct in assuming that the following is the only current workaround?

1. You're editing a piece of content in a WYSIWYG editor and want to add a link to a file from the Directus File Library.
2. Open a new browser tab.
3. Navigate to the Directus File Library
4. Find or upload the file you want to use
5. Go to the file view page
6. Open the file details sidebar
7. Right click on "Open in New Window"
8. Copy the link
9. Manually remove the `access_token` query string param because presumably this will cause access issues after that token expires if they use that (and it's unnecessary if you have public perms set to read all on files)
10. Copy _that_ edited link
11. Navigate back to the WYSIWYG editor
12. Click the area in the editor where you want to add a link to this file
13. Click "Add/Edit Link" in the toolbar
14. Paste the URL from step 10 into this field
15. Give it a title

Is there an easier way I'm missing here?

**Reply by br41nslug** [reply]
The image utility can shave a couple of steps off that list.

https://user-images.githubusercontent.com/9389634/228322031-479fad9a-1758-4b15-a93b-411c88ac8121.mp4

**Reply by nicksergeant** [reply]
Huh, yeah, bit of an interesting hack there... certainly less painful but probably way too confusing for my users, unfortunately.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
